### PR TITLE
Introduce formatted string generator

### PR DESCRIPTION
**Pull request by mfalesni**
Introduce `fauxfactory.generate()` which takes a string with formatting similar to `"".format()` but instead of inserting strings, it randomizes the values based on what is located inside the braces.

### CODE REVIEW

**Comment by mfalesni** [comment]
I wonder why does it hate line 807 in python 2.6. I think doing `# NOQA` or whatever works is OK here.

**Comment by Ichimonji10** [comment]
Using `# noqa` isn't the right thing to do here, because flake8 is not complaining. The tool which is complaining is pylint. Old versions of pylint complain about the use of `*` and `**`, and it appears that the Python 2.6 environment uses an old version of pylint. (To be expected.) The appropriate comment is `# pylint:disable=star-args`.

File: fauxfactory/__init__.py [hunk 1]
```diff
@@ -760,6 +761,64 @@ def gen_html(length=10):
...
+    optional sections. First, if needed, parameters follow starting with ``:``
+    and then separated with ``:`` if there are more than one. Then there is the
+    formatter section, starting with ``|`` and every subsequent formatter is
+    also separated with ``|``. If the formatter starts with ``.`` it will call
+    that method on the result of the preceding step and returns the result of
+    the call as a new result. Other types of formatters are not supported yet.
+
+    Example of the formatter is:
+
+    ``"test_vm_{alpha:12|.lower}"``
```
Comment (Ichimonji10) [inline fauxfactory/__init__.py]:
```
An example of the formatter is::

    'test_vm_{alpha:12|.lower}'
```

**Comment by mfalesni** [comment]
@Ichimonji10 the pylint comment as you said triggered an error in all of the python versions. I have put the `NOQA` comment there to see if that one solves the issue. I also updated the example.

**Comment by mfalesni** [comment]
Doesn't help either.

**Comment by elyezer** [comment]
I think this is a good addition and a good start. The next step would be allowing one or more generators to be specified.

We need to take care of not reinventing some behaviours implemented by other libraries like rstr [1]

[1] https://pypi.python.org/pypi/rstr/2.1.3

**Comment by elyezer** [comment]
Thinking better... I think that we can have the same behaviour without the need of any additional logic:

```python
generate("test_vm_{alpha:12|.lower}")

# is the same as

'test_vm_{0}'.format(gen_alpha(12).lower())
```

With that said, by adding this kind of generator will not bring any real improvement because we can mix the python string formatting with the generators without the overhead of parsing the generator string.

**Comment by Ichimonji10** [comment]
The `format()` method already specifies a mini-language. See examples [here](https://docs.python.org/3.4/library/string.html#format-examples). The mini-language uses the `{` and `}` characters to denote what should be interpolated. Having our own similar-but-incompatible formatting mini-language seems like a bad idea to me.

**Comment by omaciel** [comment]
@mfalesni  After reviewing the PR and after considering the ultimate goal for it, I tend to agree with what @elyezer mentioned above. I feel that we'd be re-inventing the wheel and that perhaps this could be something that you can implement in your own code to handle your specific requirement?

**Comment by mfalesni** [comment]
ok

**Pull request closed without merging**